include/pybind11/numpy.h

```diff
@@ -126,7 +135,20 @@ struct npy_api {
         NPY_FLOAT_, NPY_DOUBLE_, NPY_LONGDOUBLE_,
         NPY_CFLOAT_, NPY_CDOUBLE_, NPY_CLONGDOUBLE_,
         NPY_OBJECT_ = 17,
-        NPY_STRING_, NPY_UNICODE_, NPY_VOID_
+        NPY_STRING_, NPY_UNICODE_, NPY_VOID_,
+        // Platform-dependent normalization
+        NPY_INT8_ = NPY_BYTE_,
+        NPY_UINT8_ = NPY_UBYTE_,
+        NPY_INT16_ = NPY_SHORT_,
+        NPY_UINT16_ = NPY_USHORT_,
+        NPY_INT32_ = platform_lookup<int32_t, short, int, long>(
+            NPY_SHORT_, NPY_INT_, NPY_LONG_),
+        NPY_UINT32_ = platform_lookup<uint32_t, unsigned short, unsigned int, unsigned long>(
+            NPY_USHORT_, NPY_UINT_, NPY_ULONG_),
+        NPY_INT64_ = platform_lookup<int64_t, int, long, long long>(
+            NPY_INT_, NPY_LONG_, NPY_LONGLONG_),
+        NPY_UINT64_ = platform_lookup<uint64_t, uint, unsigned long, unsigned long long>(
```

Comment:
`uint` isn't a standard C or C++ type (even if commonly available); that should be `unsigned int`, or if you're feeling lazy, just `unsigned` (though some people strongly dislike the latter).

The `int32_t` etc. types are more portable, but should still technically be prefixed with `std::` (we should also have a `#include <cstdint>` in here to properly bring them in).

Reply:
It looks like `uint` isn't available on MSVC.

Reply:
Done. Thanks!